### PR TITLE
Dealing with undefined index if the field was empty before and the object is not dirty

### PR DESCRIPTION
**Pull request by michel-paiva**
PR solving the issue: #5895

### CODE REVIEW

File: models/DataObject/Classificationstore.php [hunk 1]
```diff
@@ -198,14 +198,18 @@ public function setLocalizedKeyValue($groupId, $keyId, $value, $language = null)
 
         if (!$this->isFieldDirty('_self')) {
             if ($this->object) {
-                $oldData = $this->items[$groupId][$keyId][$language];
```
Comment (weisswurstkanone) [inline models/DataObject/Classificationstore.php]: Would say simply adding a trailing ?? null
would be sufficient.

File: models/DataObject/Classificationstore.php [hunk 2]
```diff
@@ -198,14 +198,18 @@ public function setLocalizedKeyValue($groupId, $keyId, $value, $language = null)
...
-                $oldData = $dataDefinition->getDataForResource($oldData, $this->object);
-                $oldData = serialize($oldData);
+                $oldData = null;
+                
+                if (!isset($this->items[$groupId][$keyId][$language])) {
```
Comment (weisswurstkanone) [inline models/DataObject/Classificationstore.php]: remove the if

now we compare data which is transformed by getDataForResource with data that isn't

File: models/DataObject/Classificationstore.php [hunk 3]
```diff
@@ -198,14 +198,18 @@ public function setLocalizedKeyValue($groupId, $keyId, $value, $language = null)
...
+                    $oldData = $this->items[$groupId][$keyId][$language];
+                    $oldData = $dataDefinition->getDataForResource($oldData, $this->object);
+                    $oldData = serialize($oldData);
+                }
 
                 $newData = $dataDefinition->getDataForResource($value, $this->object);
                 $newData = serialize($newData);
 
-                if ($newData != $oldData) {
+                if (!$oldData || $newData != $oldData) {
```
Comment (weisswurstkanone) [inline models/DataObject/Classificationstore.php]: shouldn't be necessary

File: models/DataObject/Classificationstore.php (hunk 1 above)
Comment (michel-paiva) [reply models/DataObject/Classificationstore.php]: Done!

**Comment by weisswurstkanone** [comment]
thx!

**Pull request merged**